### PR TITLE
feat(Turborepo): Initial run preamble

### PR DESCRIPTION
**Pull request by gsoltis**
### Description

 - Adds packages and targets summary at the beginning of a run
 - Some minor tweaks to output to match Go logic
 - Minor bugfix to `cwrite!` macro
 - Add a `cprint!` macro for lines with mixed color formatting

### Testing Instructions

Existing test suite. We're still missing the post-run summary, so we don't have more integration tests passing, but the diff is smaller now.

Closes TURBO-1484

### CODE REVIEW

**Review by chris-olszewski — APPROVED** [review]

File: crates/turborepo-api-client/src/lib.rs [hunk 1]
```diff
@@ -100,6 +100,18 @@ pub struct APIAuth {
     pub team_slug: Option<String>,
 }
 
+impl APIAuth {
+    pub fn is_linked(&self) -> bool {
+        !self.token.is_empty()
```
Comment (mehulkar) [inline crates/turborepo-api-client/src/lib.rs:105]: Want to defer to the auth crate here? cc @Zertsov

Comment (gsoltis) [reply crates/turborepo-api-client/src/lib.rs:105]: Yes, but I think it needs some refactoring to happen. `APIAuth` might properly be owned by that crate, although we'll have to work out how that interacts with the config code.

Comment (NicholasLYang) [reply crates/turborepo-api-client/src/lib.rs:105]: I don't think this is correct. If there is no token or team id, `APIAuth` [shouldn't even be constructed](https://github.com/vercel/turbo/blob/ef2b800167d527f3fa8bc06ac98f7edaeb51b410/crates/turborepo-lib/src/run/mod.rs#L170). To check if linked we should just check if `Option<APIAuth>` is `Some`

**Review by NicholasLYang — CHANGES_REQUESTED** [review]

File: crates/turborepo-api-client/src/lib.rs (hunk 1 above)
Comment (gsoltis) [reply crates/turborepo-api-client/src/lib.rs:105]: Ah ok. That makes sense, if having an `APIAuth` implies that we at least have a token and a team id, then that works

Comment (gsoltis) [reply crates/turborepo-api-client/src/lib.rs]: Switched to `is_linked = api_auth.is_some()`

**Review by NicholasLYang — APPROVED** [review]

**Pull request merged**